.github/workflows/kitchen-fips.yml

```diff
@@ -0,0 +1,77 @@
+        ruby: ['3.1']
+    runs-on: ${{ matrix.os }}
+    steps:
+    - uses: actions/checkout@v3
+    - name: 'Install Chef/Ohai from Omnitruck'
+      id: install_chef
+      run: |
+        . { Invoke-WebRequest -useb https://omnitruck.chef.io/install.ps1 } | Invoke-Expression; Install-Project -project chef -channel current
+        $env:PATH = "C:\opscode\chef\bin;C:\opscode\chef\embedded\bin;" + $env:PATH
+        chef-client -v
+        ohai -v
+        rake --version
+        bundle -v
+    - name: 'Upgrade Chef/Ohai via Appbundler'
+      id: upgrade
+      run: |
+        $env:PATH = "C:\opscode\chef\bin;C:\opscode\chef\embedded\bin;" + $env:PATH
+        $env:OHAI_VERSION = ( Select-String -Path .\Gemfile.lock -Pattern '(?<=ohai \()\d.*(?=\))' | ForEach-Object { $_.Matches[0].Value } )
+
+        # The chef-client installer does not put the file 'ansidecl.h' down in the correct location
+        # This leads to failures during testing. Moving that file to its correct position here.
+        # Another example of 'bad' that needs to be corrected
+        $output = gci -path C:\opscode\ -file ansidecl.h -Recurse
+
+        # As of Ruby 3.1, there are 3 ansidecl.h files found in the opscode path
+        # Grabbing the first (and shortest) path found is a bit of a :fingers-crossed: but
+        # making the leap that ansidecl.h isn't going to vary in a way that will fail
+        # subtly.
+        if ($output -is [Array]) { $output = $output[0] }
+
+        $target_path = $($output.Directory.Parent.FullName + "\x86_64-w64-mingw32\include")
+        Move-Item -Path $output.FullName -Destination $target_path
+
+        # if a different version of ffi-yajl is installed, then libyajl2 needs to be reinstalled
+        # so that libyajldll.a is present in the intermediate build step. bundler seems to skip
+        # libyajl2 build if already present. gem install seems to build anyway.
+        gem uninstall -I libyajl2
+
+        gem install appbundler appbundle-updater --no-doc
+        If ($lastexitcode -ne 0) { Exit $lastexitcode }
+        appbundle-updater chef chef $env:GITHUB_SHA --tarball --github $env:GITHUB_REPOSITORY
+        If ($lastexitcode -ne 0) { Exit $lastexitcode }
+        Write-Output "Installed Chef / Ohai release:"
+        ohai -v
+        If ($lastexitcode -ne 0) { Exit $lastexitcode }
+        chef-client -v
+        If ($lastexitcode -ne 0) { Exit $lastexitcode }
+    - name: 'Enable FIPS via windows registry' #FIPS needs to be enabled after installation of Chef, otherwise it fails due to Cryptography error
+      run: |
+        $RegistryPath = 'HKLM:\SYSTEM\CurrentControlSet\Control\Lsa\FipsAlgorithmPolicy'
+        $Key = 'Enabled'
+        $Value = '1'
+        If (-NOT (Test-Path $RegistryPath)) {
+          throw "Registry path to activate FIPS does not exist."
+        }
+        Set-ItemProperty -Path $RegistryPath -Name $Key -Value $Value
+        if (-not $?) { throw "Failed to enable FIPS mode." }
```

Comment:
Hi, this needs to be updated. The "throw" command is a terminating error - meaning the script will exit right then if the registry key is not found. Use this instead:

```powershell
$KeyPath = "HKLM:\SYSTEM\CurrentControlSet\Control\Lsa\FipsAlgorithmPolicy"  
$ValueName = "Enabled"  
$ValueData = "1"  
try{  
    Get-ItemProperty -Path $KeyPath -Name $valueName -ErrorAction Stop  
}  
catch [System.Management.Automation.ItemNotFoundException] {  
    New-Item -Path $KeyPath -Force  
    New-ItemProperty -Path $KeyPath -Name $ValueName -Value $ValueData -Force  
}  
catch {  
    New-ItemProperty -Path $KeyPath -Name $ValueName -Value $ValueData -Type String -Force  
}
```

Reply:
@johnmccrae  If FIPS enabling has failed, there is no need to execute code further that's why the 'throw' was used.